config/main.py

```diff
@@ -2661,6 +2662,25 @@ def remove(ctx, interface_name, ip_addr):
         table_name = get_interface_table_name(interface_name)
         if table_name == "":
             ctx.fail("'interface_name' is not valid. Valid names [Ethernet/PortChannel/Vlan/Loopback]")
+        interface_dependent = interface_ipaddr_dependent_on_interface(config_db, interface_name)
+        # If we deleting the last IP entry of the interface, check whether a static route present for the RIF
+        # before deleting the entry and also the RIF.
+        if len(interface_dependent) == 1 and interface_dependent[0][1] == ip_addr:
+            # Check both IPv4 and IPv6 routes.
+            ip_versions = [ "ip", "ipv6"]
+            for ip_ver in ip_versions:
+                # Compete the command and ask Zebra to return the routes.
+                # Scopes of all VRFs will be checked.
+                cmd = "show {} route vrf all".format(ip_ver)
```

Comment:
why `all`? why not just `show ip route static`. Lets optimize this as best as possible since this could potentially introduce further delays in this command.

Reply:
`vrf all` is required to check scopes of all the VRFs present in the system, because command `show ip route` shows routes only in scope of default VRF.
Good point regarding static prefix, I have updated and retested the changes. Thank you.
Please, take a look on the difference between outputs of commands `show ip route static` and `show ip route vrf all static`. That's why I have added `vrf all`:
```
admin@sonic:~$ show ip route vrf all static
Codes: K - kernel route, C - connected, S - static, R - RIP,
       O - OSPF, I - IS-IS, B - BGP, E - EIGRP, N - NHRP,
       T - Table, v - VNC, V - VNC-Direct, A - Babel, D - SHARP,
       F - PBR, f - OpenFabric,
       > - selected route, * - FIB route, q - queued, r - rejected, b - backup

VRF Vrf11:
S   20.0.0.144/32 [1/0] is directly connected, Ethernet68 inactive, weight 1, 00:00:18

VRF default:
S>* 0.0.0.0/0 [200/0] via 192.168.111.3, eth0, weight 1, 23:09:50
S   20.0.0.1/32 [1/0] is directly connected, Ethernet64 inactive, weight 1, 00:01:14
                      is directly connected, Ethernet68 (vrf Vrf11) inactive, weight 1, 00:01:14

admin@sonic:~$ show ip route static
Codes: K - kernel route, C - connected, S - static, R - RIP,
       O - OSPF, I - IS-IS, B - BGP, E - EIGRP, N - NHRP,
       T - Table, v - VNC, V - VNC-Direct, A - Babel, D - SHARP,
       F - PBR, f - OpenFabric,
       > - selected route, * - FIB route, q - queued, r - rejected, b - backup

S>* 0.0.0.0/0 [200/0] via 192.168.111.3, eth0, weight 1, 23:10:01
S   20.0.0.1/32 [1/0] is directly connected, Ethernet64 inactive, weight 1, 00:01:25
                      is directly connected, Ethernet68 (vrf Vrf11) inactive, weight 1, 00:01:25
```